tests/regression/test_moore_spence.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+from firedrake import *
+from firedrake.petsc import PETSc
+from slepc4py import SLEPc
```

Comment:
This test fails because the builders don't currently install slepc. I can fix that, but we should still fail gracefully if the user didn't install slepc. Rather than importing at the top level, can you do:

```
    try:
        from slepc4py import SLEPc
    except ImportError:
        pytest.skip(msg="SLEPc unavailable, skipping eigenvalue test")
```

inside the test itself.